crates/swc_compiler_base/src/lib.rs

```diff
@@ -37,17 +37,28 @@ pub struct TransformOutput {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub output: Option<String>,
+
+    pub diagnostics: std::vec::Vec<String>,
 }
 
 #[cfg(not(feature = "node"))]
 #[derive(Debug, Serialize)]
 pub struct TransformOutput {
     pub code: String,
+
     #[serde(skip_serializing_if = "Option::is_none")]
     pub map: Option<String>,
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub output: Option<String>,
+
+    pub diagnostics: std::vec::Vec<String>,
```

Comment:
This is a breaking change anyway

Reply:
`TransformOutput` may need include [diagnostics](https://github.com/swc-project/swc/blob/main/crates/swc_common/src/errors/diagnostic.rs#L29-L48) struct.
What do you think?

Reply:
This should be json string I think, so it's fine. But we should not render the diagnostic in the current way.

Reply:
> But we should not render the diagnostic in the current way.

You means [here](https://github.com/swc-project/swc/blob/5efe61828a5372b066dc473de078be115157bed8/crates/swc_error_reporters/src/lib.rs#L181-L189)?

Reply:
Can we have an option to set the formatting of error messages? Like `--message-format=json` of various tools.

This may be out of the scope of the PR, but I think we cannot render it (`miette`/`PrettyEmitter`) or serialize it (json) without such option

Reply:
Hopefully in `HandlerOpts`

Reply:
Got it, i will implement it recently.